src/baseline_bots/bots/smart_order_accepter_bot.py

```diff
@@ -48,23 +48,64 @@ class SmartOrderAccepterBot(DipnetBot):
     """
 
     def __init__(
-        self, power_name, game, discount_factor=0.5, test_mode=False, stance_type="A"
+        self,
+        power_name,
+        game,
+        discount_factor=0.5,
+        test_mode=False,
+        stance_type="A",
+        aggressiveness="M",
     ) -> None:
         """
         :param power_name: The name of the power
         :param game: Game object
         :param discount_factor: discount factor for ActionBasedStance
         :param test_mode: indicates if this bot is to be executed in test mode or not. In test_mode, async function `send_message` will not be used.
         :param stance_type: indicates if this bot should use ActionBasedStance (A) or ScoreBasedStance (S)
+        :param aggressiveness: indicates if this bot should be aggressive (A), moderate (M) or friendly (F). Valid only if stance type is action-based
         """
         super().__init__(power_name, game)
         self.alliance_props_sent = False
         self.discount_factor = discount_factor
         self.stance_type = stance_type
+        self.aggressiveness = aggressiveness
         if self.stance_type == "A":
-            self.stance = ActionBasedStance(
-                power_name, game, discount_factor=self.discount_factor
-            )
+            if self.aggressiveness == "A":
+                self.stance = ActionBasedStance(
+                    power_name,
+                    game,
+                    invasion_coef=2.0,
+                    conflict_coef=1.0,
+                    invasive_support_coef=2.0,
+                    conflict_support_coef=1.0,
+                    friendly_coef=1.0,
+                    unrealized_coef=1.0,
+                    discount_factor=self.discount_factor,
+                )
+            elif self.aggressiveness == "M":
+                self.stance = ActionBasedStance(
+                    power_name,
+                    game,
+                    invasion_coef=1.0,
+                    conflict_coef=0.5,
+                    invasive_support_coef=1.0,
+                    conflict_support_coef=0.5,
+                    friendly_coef=1.0,
+                    unrealized_coef=1.0,
+                    discount_factor=self.discount_factor,
+                )
+            elif self.aggressiveness == "F":
+                self.stance = ActionBasedStance(
+                    power_name,
+                    game,
+                    invasion_coef=0.5,
+                    conflict_coef=0.25,
+                    invasive_support_coef=0.5,
+                    conflict_support_coef=0.25,
+                    friendly_coef=0.5,
+                    unrealized_coef=1.0,
+                    discount_factor=self.discount_factor,
+                )
```

Comment:
There should be a final `else` clause here to throw an error if the option is unrecognized.